examples/semi_supervised/plot_self_training_performance.py

```diff
@@ -44,14 +40,13 @@
 X, y = shuffle(X, y, random_state=42)
 y_true = y.copy()
 
-for i in range(3, -1, -1):
-    lim = x_values[i]
-    y[lim:] = -1
+for i in range(x_values.shape[0]-1, -1, -1):
```

Comment:
you'll need a space in-between ``_values.shape[0] - 1``

Reply:
Weird that flake8 didn't pick up on this